### ISSUE TITLE
Confusion around current_view/set_view

### ISSUE THREAD

**Issue by pvcraven** [pvcraven/arcade#662]
People are setting window.current_view instead of using window.show_view() leading to some interesting bugs
We should probably make current_view into read only @property
It means removing the events from the next view will not work and you are stuck with 2 views getting events.

**Comment by SirGnip** [comment]
Would simply changing the name of `current_view` to `_current_view` be enough?  This is [PEP 8's](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables) recommendation for non-public members.  Python doesn't enforce it, but some IDE's and linters flag it.

**Comment by einarf** [comment]
Yes, but users are accessing `current_view` to check the current view state in their game logic. So wrapping it in a property getter/setter like this is probably the safest.

```python
class Window:
    def __init__(self):
        self._current_view = None

   @property
   def current_view(self):
        return self._current_view

   @current_view.setter
   def current_view(self. view):
       self.show_view(view)

**Comment by SirGnip** [comment]
Ah.  That makes sense.  Thx.

If the new implementation is similar to above, how does that mean the "removing of events from the next view" will not work?  `self.current_view` in `show_view()` just needs to be changed to `self._current_view`?  What am I missing?

**Comment by einarf** [comment]
This was poorly worded by me and pasted in by Paul 😄 It was describing the bug people experienced when setting the value using the property in the current code base.

What I meant was `show_view()` removes handlers from the previous view. People were modifying `current_view` directly causing handlers not to be removed so they would have multiple views rendering and taking events at the same time.

**Comment by SirGnip** [comment]
:)  No worries. Thx for the clarification.  +1 from me on making self.current_view private.

However, using the code snippet above would mean there would be two ways for client code to change views... each of them effectively accomplishing the same thing which is confusing.

`self.window.current_view = new_view`  # Approach A
...and...
`self.window.show_view(new_view)`  # Approach B

I think we can either...
* Make _current_view private, do the read-only @property work as in the code above and move the logic of `show_view()` into @current_view.setter.  Users ONLY set new views using Approach A above.  This would break existing apps.  Or...
* Make _current_view private and leave the current implementation of `show_view()` mostly as-is.  No properties.  Users ONLY set new views using Approach B.

I don't have much of opinion on which approach is better, as long as ["There should be one-- and preferably only one --obvious way to do it."](https://www.python.org/dev/peps/pep-0020/)

**Comment by einarf** [comment]
I think because `current_view` used to be a public property that was settable and gettable and there are instances where setting this property would not break anything, the most important thing is to not break backwards compatibility (Even if there now is two ways of doing it).

We're at the end of a 10 day game jam here and have been following what all the participants have been doing, and this was one of the issues that broke their games badly.

**Comment by SirGnip** [comment]
(The game jam looks slick... wish I would have seen the announcement earlier! Can't wait to see what comes from it!)

Yes, `current_view` being public causes problems.  It needs to be fixed.  Backwards compatibility is also very important.  And, with arcade being a library that is often used for teaching newcomers to Python... having a clear, obvious interface is also very important so as to reduce the confusion experienced by newcomers.
This discussion makes me feel like Approach B is the best.  It makes `current_view` private (`_current_view`) and is backwards compatible with correct code (apps that use `.show_view()` approach will not have to change).

**Comment by pvcraven** [comment]
After pondering this a while, I originally thought about making both a getter and setter. Wrote the code. Decided I didn't like having a setter, as I don't feel they should have that much logic behind them. So I'm thinking read-only property.

**Comment by pvcraven** [comment]
Released in 2.4.
